doc: gsg: macOS: Do not require wget, use curl instead

carlescufi opened the pull request:
macOS ships with curl by default. There's no need to add additional tools to the installation when a perfectly good one ships with the OS.

carlescufi committed:
doc: gsg: macOS: Do not require wget, use curl instead

macOS ships with curl by default. There's no need to add additional
tools to the installation when a perfectly good one ships with the OS.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

kartben reviewed (changes requested):
well spotted!

Please make sure https://github.com/zephyrproject-rtos/zephyr/blob/bb5098da082b3e5e959918a43a8810843754450c/doc/develop/toolchains/zephyr_sdk.rst#L133-L135 is also updated. .... but it feels like we may actually want to directly `..include` snippets from this file it in the getting started guide, looks like a lot of duplicated content (and I now remember on or the other of these two files not being properly updated during last update to 0.16.3).

There are a couple other occurrences of wget usage in the tree but I think it's fine to assume people getting that far would know how to install wget, if they haven't already.

carlescufi commented:
> is also updated. .... but it feels like we may actually want to directly `..include` snippets from this file it in the getting started guide, looks like a lot of duplicated content (and I now remember on or the other of these two files not being properly updated during last update to 0.16.3).

OK, I will give this a go, not sure if it'll work out with tabs and so on?

kartben commented:
> > is also updated. .... but it feels like we may actually want to directly `..literalinclude` snippets from this file it in the getting started guide, looks like a lot of duplicated content (and I now remember on or the other of these two files not being properly updated during last update to 0.16.3).
> 
> OK, I will give this a go, not sure if it'll work out with tabs and so on?

There's no reason why it shouldn't work, even in tab content, as long as nested indentation is respected. And you'll need the start-after/end-before markers to identify which portions to include of course

kartben commented:
> > is also updated. .... but it feels like we may actually want to directly `..literalinclude` snippets from this file it in the getting started guide, looks like a lot of duplicated content (and I now remember on or the other of these two files not being properly updated during last update to 0.16.3).
> 
> OK, I will give this a go, not sure if it'll work out with tabs and so on?

There's no reason why it shouldn't work, even in tab content, as long as nested indentation is respected. And you'll need the start-after/end-before markers to identify which portions to include of course

carlescufi committed:
doc: Reuse the same Zephyr SDK installation instructions

Instead of keeping 2 copies of the same Zephyr SDK installation
instructions, reuse a single copy by including it in the getting stated
guide from the Zephyr SDK doc page.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
> > > is also updated. .... but it feels like we may actually want to directly `..include` snippets from this file it in the getting started guide, looks like a lot of duplicated content (and I now remember on or the other of these two files not being properly updated during last update to 0.16.3).
> > 
> > 
> > OK, I will give this a go, not sure if it'll work out with tabs and so on?
> 
> There's no reason why it shouldn't work, even in tab content, as long as nested indentation is respected. And you'll need the start-after/end-before markers to identify which portions to include of course

I think it should work now @kartben. Please take another look.

kartben commented on doc/develop/toolchains/zephyr_sdk.rst:
```diff
@@ -67,153 +67,173 @@ the recommended version for the corresponding Zephyr version.
+          SDK releases.
+
+.. note:: If you want to uninstall the SDK, remove the directory where you installed it.
+          If you relocate the SDK directory, you need to re-run the setup script.
```
this is redundant as already mentioned at the end of the install instructions? I think it should be removed from one or the other(s), but I don't have a strong opinion as to which

carlescufi replied:
Fixed

kartben commented on doc/develop/toolchains/zephyr_sdk.rst:
```diff
@@ -67,153 +67,173 @@ the recommended version for the corresponding Zephyr version.
+          if needed; the `Zephyr SDK Releases`_ page contains all available
+          SDK releases.
+
+.. note:: If you want to uninstall the SDK, remove the directory where you installed it.
```
```suggestion
.. note:: If you want to uninstall the SDK, you may simply remove the directory where you installed it.
```

carlescufi replied:
Thanks!

kartben reviewed (changes requested):
Sphinx (and negative diffs!) ftw!

kartben commented:
@carlescufi looks like we should try doing something similar with "Install Dependencies" steps as we tend to get disconnects between what's in the getting started and what's in `installation_*.rst` files (see ex. #66461)
